tutor/commands/dev.py

```diff
@@ -23,6 +24,13 @@ def __init__(self, root: str, config: Config):
             tutor_env.pathjoin(self.root, "local", "docker-compose.override.yml"),
             tutor_env.pathjoin(self.root, "dev", "docker-compose.override.yml"),
         ]
+
+        if not utils.is_docker_rootless():
```

Comment:
Let's not create a separate file just for this use case. Instead, the dev/docker-compose.yml file should include a section:

```
  {% if RUN_ELASTICSEARCH -%}
  elasticsearch:
    ulimits: {}
  {%- endif %}
```

Reply:
```
elasticsearch:
    ulimits: {}
```

Won't work and throws the same exception. 

Setting the `ulimit` to `False` or anything else won't matter. To address your idea, the `if` statement should be added within the production file. But I think we're converging to an agreement!